security/pfSense-pkg-acme/files/usr/local/pkg/acme/acme.inc

```diff
@@ -1370,10 +1370,13 @@ function & get_certificate($name) {
 			echo "\n{$cert['prv']}";
 			$desc = "Acme: Add new certificate & key.";
 			write_config($desc);
-		} elseif (   $certificate['keylength'] == 'custom'
-				  && $cert['prv'] != $certificate['keypaste']) {
+		} elseif (($certificate['keylength'] == 'custom') && ($cert['prv'] != $certificate['keypaste'])) {
 			echo "\n getCertificatePSK updating custom key";
 			$cert['prv'] = $certificate['keypaste']; // (already base64-encoded)
+		} elseif ($certificate['keylength'] != 'custom') {
```

Comment:
What's the logic behind this test? It would appear to make a new key any time the function is run on an existing certificate rather than keeping the existing key.

It should only generate a new key if there is a change in the `keylength` value.

Reply:
Updated
now it compares the `keylength` with the `$cert['prv']`